### PR TITLE
[PIO-1] Make SSL and authKey param authentication optional

### PR DESCRIPTION
**Pull request by chanlee514**

### CODE REVIEW

**Comment by chanlee514** [comment]
[PIO-1]

**Comment by dszeto** [comment]
Testing JIRA integration.

File: core/src/main/scala/io/prediction/workflow/CreateServer.scala [hunk 1]
```diff
@@ -291,9 +291,12 @@ class MasterActor (
   var sprayHttpListener: Option[ActorRef] = None
   var currentServerActor: Option[ActorRef] = None
   var retry = 3
+  val serverConfig = ConfigFactory.load("server.conf")
+  val sslEnforced = serverConfig.getBoolean("io.prediction.server.ssl-enforced")
+  val protocol = if (sslEnforced) "https://" else "http://"
 
   def undeploy(ip: String, port: Int): Unit = {
-    val serverUrl = s"https://${ip}:${port}"
+    val serverUrl = protocol + s"${ip}:${port}"
```
Comment (dszeto) [inline core/src/main/scala/io/prediction/workflow/CreateServer.scala]: @chanlee514 How about `s"$protocol${ip}:${port}"`?

File: core/src/main/scala/io/prediction/workflow/CreateServer.scala [hunk 2]
```diff
@@ -377,7 +380,7 @@ class MasterActor (
           s"${manifest.version}. Abort reloading.")
       }
     case x: Http.Bound =>
-      val serverUrl = s"https://${sc.ip}:${sc.port}"
+      val serverUrl = protocol + s"${sc.ip}:${sc.port}"
```
Comment (dszeto) [inline core/src/main/scala/io/prediction/workflow/CreateServer.scala]: Same as above.

**Comment by dszeto** [comment]
Inline. Otherwise LGTM.

**Pull request closed without merging**